sample/ws-chat-server.c

```diff
@@ -202,6 +202,8 @@ on_html(struct evhttp_request *req, void *arg)
 
 err:
 	evhttp_send_error(req, HTTP_NOTFOUND, NULL);
+	if (fd>=0)
+		close(fd);
```

Comment:
`fd` should be closed by the `evbuffer` (it is passed to the `evbuffer_add_file` and `evbuffer` should close it internally)

Reply:
If `fstat` fails, it will go to `err` and skip `evbuffer`, so it is nesseary to close `fd` explicitly in `err`, just similar to [sample/http-server.c (line: 337)](https://github.com/libevent/libevent/blob/master/sample/http-server.c)

Reply:
Yes, missed the `err` label, you are right.